source/math/FloatComplex.ooc

```diff
@@ -60,59 +59,6 @@ FloatComplex: cover {
 	rootOfUnity: static func (n: Int, k:= 1) -> This {
 		This new(0, 2 * k * PI / n) exponential()
 	}
-	discreteFourierTransform: static func (input: HeapVector<This>) -> HeapVector<This> {
-		result := HeapVector<This> new(input capacity)
-		for (i in 0..(input capacity)) {
-			for (j in 0..(input capacity))
-				result[i] = result[i] + input[j] * FloatComplex rootOfUnity(input capacity, -i * j)
-		}
-		result
-	}
-	inverseDiscreteFourierTransform: static func (input: HeapVector<This>) -> HeapVector<This> {
-		result := HeapVector<This> new(input capacity)
-		for (i in 0..(input capacity)) {
-			result[i] = input[i] conjugate
-		}
-		result = FloatComplex discreteFourierTransform(result)
-		for (i in 0..(result capacity)) {
-			result[i] = (result[i] conjugate) / (input capacity)
-		}
-		result
-	}
-	fastFourierTransform: static func (input: HeapVector<This>) -> HeapVector<This> {
-		result := HeapVector<This> new(input capacity)
-		if (input capacity == 1)
-			result = input
-		else {
-			halfLength: Int = input capacity / 2
-			evenInput := HeapVector<This> new(halfLength)
-			oddInput := HeapVector<This> new(halfLength)
-			for (i in 0..halfLength) {
-				evenInput[i] = input[2 * i]
-				oddInput[i] = input[2 * i + 1]
-			}
-			evenOutput := FloatComplex fastFourierTransform(evenInput)
-			oddOutput := FloatComplex fastFourierTransform(oddInput)
-			root: This
-			for (i in 0..halfLength) {
-				root = FloatComplex rootOfUnity(input capacity, -i)
-				result[i] = evenOutput[i] + root * oddOutput[i]
-				result[halfLength + i] = evenOutput[i] - root * oddOutput[i]
-			}
-		}
-		result
-	}
-	inverseFastFourierTransform: static func (input: HeapVector<This>) -> HeapVector<This> {
-		result := HeapVector<This> new(input capacity)
-		for (i in 0..(input capacity)) {
-			result[i] = input[i] conjugate
-		}
-		result = FloatComplex fastFourierTransform(result)
-		for (i in 0..(result capacity)) {
-			result[i] = (result[i] conjugate) / (input capacity)
-		}
-		result
-	}
```

Comment:
The idea was that we should move `FloatVector` and `FloatPoint2DList` and `FloatComplexList` or whatever their names are to `ooc-math`, and then move the Fourer transform methods into `FloatComplexList`. I think.